src/units_for_slot_test.rs

```diff
@@ -242,8 +260,463 @@ mod units_for_slot_tests {
+        channel.spec.ad_units = vec![];
+
+        let config = Config::new(None, "development").expect("should get config");
+        let mock_client = MockClient::init(
+            vec![mock_cache_campaign(channel.clone(), Status::Active)],
+            vec![],
+            None,
+        )
+        .await;
+
+        let mock_cache = Cache::initialize(mock_client).await;
+
+        let ad_units: Vec<AdUnit> = get_supermarket_ad_units();
+        let mock_slot = get_supermarket_ad_slot(&rules, &categories);
+
+        Mock::given(method("GET"))
+            .and(path("/market/units"))
+            .respond_with(ResponseTemplate::new(200).set_body_json(&ad_units))
+            .mount(&server)
+            .await;
+
+        Mock::given(method("GET"))
+            .and(path(format!("/market/slots/{}", mock_slot.slot.ipfs)))
+            .respond_with(ResponseTemplate::new(200).set_body_json(&mock_slot))
+            .mount(&server)
+            .await;
+        let expected_response = get_expected_response(vec![]);
+        let request = Request::get(format!(
+            "/units-for-slot/{}?depositAsset=0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359",
+            mock_slot.slot.ipfs
+        ))
+        .body(Body::empty())
+        .unwrap();
+
+        let actual_response =
+            get_units_for_slot(&logger, market.clone(), &config, &mock_cache, request)
+                .await
+                .expect("call shouldn't fail with provided data");
+
+        assert_eq!(http::StatusCode::OK, actual_response.status());
+
+        let units_for_slot: UnitsForSlotResponse =
+            serde_json::from_slice(&hyper::body::to_bytes(actual_response).await.unwrap())
+                .expect("Should deserialize");
+
+        // FIXME: Enable assert when issue #340 is solved: https://github.com/AdExNetwork/adex-validator-stack-rust/issues/340
+        // assert_eq!(expected_response.targeting_input_base, units_for_slot.targeting_input_base);
+
+        assert_eq!(
+            expected_response.campaigns.len(),
+            units_for_slot.campaigns.len()
+        );
+        assert_eq!(expected_response.campaigns, units_for_slot.campaigns);
+        assert_eq!(
+            expected_response.fallback_unit,
+            units_for_slot.fallback_unit
+        );
+    }
+
+    #[tokio::test]
+    async fn price_less_than_min_per_impression() {
+        let logger = discard_logger();
+
+        let server = MockServer::start().await;
+
+        let market = Arc::new(
+            MarketApi::new(server.uri() + "/market", logger.clone())
+                .expect("should create market instance"),
+        );
+
+        let categories: [&str; 3] = ["IAB3", "IAB13-7", "IAB5"];
+        let rules = get_mock_rules(&categories);
+        let mut channel = mock_channel(&rules);
+        channel.spec.min_per_impression = 1_000_000.into();
+        let config = Config::new(None, "development").expect("should get config");
```

Comment:
these config crations could be replaced with `crate::config::DEVELOPMENT.clone()`